vpr/src/place/net_cost_handler.cpp

```diff
@@ -88,14 +94,33 @@ static vtr::vector<ClusterNetId, NetUpdateState> bb_updated_before; // [0...clus
  * layer_ts_bb_coord_new are used.
  */
 
-/* [0...cluster_ctx.clb_nlist.nets().size()-1] -> 3D bounding box*/
-static vtr::vector<ClusterNetId, t_bb> ts_bb_coord_new, ts_bb_edge_new;
-/* [0...cluster_ctx.clb_nlist.nets().size()-1][0...num_layers-1] -> 2D bonding box on a layer*/
-static vtr::vector<ClusterNetId, std::vector<t_2D_bb>> layer_ts_bb_edge_new, layer_ts_bb_coord_new;
-/* [0...cluster_ctx.clb_nlist.nets().size()-1][0...num_layers-1] -> number of sink pins on a layer*/
-static vtr::Matrix<int> ts_layer_sink_pin_count;
-/* [0...num_afftected_nets] -> net_id of the affected nets */
-static std::vector<ClusterNetId> ts_nets_to_update;
+struct TSInfo{
+    /* [0...cluster_ctx.clb_nlist.nets().size()-1] -> 3D bounding box*/
+    vtr::vector<ClusterNetId, t_bb> ts_bb_coord_new, ts_bb_edge_new;
+    /* [0...cluster_ctx.clb_nlist.nets().size()-1][0...num_layers-1] -> 2D bonding box on a layer*/
+    vtr::vector<ClusterNetId, std::vector<t_2D_bb>> layer_ts_bb_edge_new, layer_ts_bb_coord_new;
+    /* [0...cluster_ctx.clb_nlist.nets().size()-1][0...num_layers-1] -> number of sink pins on a layer*/
+    vtr::Matrix<int> ts_layer_sink_pin_count;
+    /* [0...num_afftected_nets] -> net_id of the affected nets */
+    std::vector<ClusterNetId> ts_nets_to_update;
+};
+static struct TSInfo ts_info;
+
+/**
+ * @brief This class is used to hide control flows needed to distinguish 2d and 3d placement
+ */
+class BB2D3DControlFlow {
```

Comment:
The classes and structs defined in this file are polluting the namespace. They either need to be put in the anonymous namespace or made static.

Although it may sound crazy, in the future (in a completely separate cpp file) someone may may a TSInfo struct (they just happen to use the same name) and they will get awful linker errors. Whenever a class/struct/function are declared in a cpp file they should be made static / put in the anonymous namespace to prevent this issue.

Reply:
We can put all the functions in `net_cost_handler` into a separate namespace. We are also following the same practice for router lookahead utilities. If you think this is a standard practice, we can at least enforce it for the new code being added to VTR.

Reply:
I think it should be enforced for new code. and It does not have to be a separate namespace; it need only be static / put in the anonymous namespace. This is common practice for very very large code bases where polluting the global space can be a real problem.

Reply:
For file scope variables, static makes sense. For struct / class definitions: probably put in anonymous namespace if you want to hide them.